src/libs/antares/study/study.cpp

```diff
@@ -547,6 +545,63 @@ void Study::getNumberOfCores(const bool forceParallel, const uint nbYearsParalle
         else
             buildNewSet = false;
     } // End of loop over years
+    
+    return setsOfParallelYears;
+
+}
+
+void Study::setGUIMinMaxNbYearsInParallel(const uint min, const uint max){
+    minNbYearsInParallel_save = min;
+    maxNbYearsInParallel_save = max;
+}
+
+bool Study::allSetsParallelYearsHaveSameSize(const std::vector<std::vector<uint>> setsOfParallelYears){
+
+    if (parameters.initialReservoirLevels.iniLevels == Antares::Data::irlHotStart
+        && setsOfParallelYears.size() && maxNbYearsInParallel > 1)
+    {
+        uint currentSetSize = (uint)setsOfParallelYears[0].size();
+        if (setsOfParallelYears.size() > 1)
+        {
+            for (uint s = 1; s < setsOfParallelYears.size(); s++)
+            {
+                if (setsOfParallelYears[s].size() != currentSetSize)
+                {
+                    return false;
+                }
+            }
+        }
+    } // End if hot start
+    return true;
+}
+
+void Study::computeNumberOfCores(const bool forceParallel, const uint nbYearsParallelForced)
+{
+    /*
+            Getting the number of parallel years based on the number
+            of cores level.
+            This number is limited by the smallest refresh span (if at least
+            one type of time series is generated)
+    */
+    computeRawNbParallelYear();
+
+    maxNbYearsInParallel = getNbYearsParallelRaw();
+
+    // In case solver option '--force-parallel n' is used, previous computation is overridden.
+    if (forceParallel)
+        maxNbYearsInParallel = nbYearsParallelForced;
```

Comment:
Don't call `getNbYearsParallelRaw` in this case, since its results are overridden ?